sdk-tests/src/test/java/io/dapr/it/testcontainers/workflows/crossapp/WorkflowsCrossAppCallActivityIT.java

```diff
@@ -0,0 +1,267 @@
+        .withSubscription(new Subscription("local", "pubsub", "topic", "/events"))
+        .withLogConsumer(outputFrame -> System.out.println("APP2: " + outputFrame.getUtf8String()))
+        .waitingFor(Wait.forHttp("/v1.0/healthz/outbound")
+            .forPort(3501)
+            .forStatusCode(200)
+            .forStatusCode(204));
+    
+    APP3_CONTAINER = new DaprContainer(DAPR_RUNTIME_IMAGE_TAG)
+        .withAppName("app3")
+        .withNetwork(DAPR_NETWORK)
+        .withNetworkAliases("app3-sidecar")
+        .withPlacementContainer(sharedPlacementContainer)
+        .withSchedulerContainer(sharedSchedulerContainer)
+        .withCustomHttpPort(3502)
+        .withCustomGrpcPort(50003)
+        .withAppChannelAddress("main-workflow-sidecar:3500")
+        .withDaprLogLevel(DaprLogLevel.DEBUG)
+        .withComponent(new Component("kvstore", "state.in-memory", "v1",
+            Map.of("actorStateStore", "true")))
+        .withComponent(new Component("pubsub", "pubsub.in-memory", "v1", Collections.emptyMap()))
+        .withSubscription(new Subscription("local", "pubsub", "topic", "/events"))
+        .withLogConsumer(outputFrame -> System.out.println("APP3: " + outputFrame.getUtf8String()))
+        .waitingFor(Wait.forHttp("/v1.0/healthz/outbound")
+            .forPort(3502)
+            .forStatusCode(200)
+            .forStatusCode(204));
+    
+    // Start the Dapr containers
+    MAIN_WORKFLOW_CONTAINER.start();
+    APP2_CONTAINER.start();
+    APP3_CONTAINER.start();
+
+    // Start crossapp worker (connects to MAIN_WORKFLOW_CONTAINER)
+    crossappWorker = new GenericContainer<>("openjdk:17-jdk-slim")
+        .withCopyFileToContainer(MountableFile.forHostPath("target/test-classes"), "/app/classes")
+        .withCopyFileToContainer(MountableFile.forHostPath("target/dependency"), "/app/libs")
+        .withWorkingDirectory("/app")
+        .withCommand("java", "-cp", "/app/classes:/app/libs/*", 
+                     "-Ddapr.app.id=crossapp-worker",
+                     "-Ddapr.grpc.endpoint=main-workflow-sidecar:50001",
+                     "-Ddapr.http.endpoint=main-workflow-sidecar:3500",
+                     "io.dapr.it.testcontainers.workflows.crossapp.CrossAppWorker")
+        .withNetwork(DAPR_NETWORK)
+        .waitingFor(Wait.forLogMessage(".*CrossAppWorker started.*", 1))
+        .withLogConsumer(outputFrame -> System.out.println("CrossAppWorker: " + outputFrame.getUtf8String()));
+    
+    // Start app2 worker (connects to APP2_CONTAINER)
+    app2Worker = new GenericContainer<>("openjdk:17-jdk-slim")
+        .withCopyFileToContainer(MountableFile.forHostPath("target/test-classes"), "/app/classes")
+        .withCopyFileToContainer(MountableFile.forHostPath("target/dependency"), "/app/libs")
+        .withWorkingDirectory("/app")
+        .withCommand("java", "-cp", "/app/classes:/app/libs/*", 
+                     "-Ddapr.app.id=app2",
+                     "-Ddapr.grpc.endpoint=app2-sidecar:50002",
```

Comment:
@cicoyle here you can use the Testcontainers mapped (random) port instead of relying on a custom port number.